hibernate-core/src/main/java/org/hibernate/type/ComponentType.java

```diff
@@ -472,7 +473,12 @@ public String toLoggableString(Object value, SessionFactoryImplementor factory)
 		Map<String, String> result = new HashMap<>();
 		Object[] values = getPropertyValues( value, entityMode );
```

Comment:
I'm not sure how lazy embedded values work. Can the embedded value as a whole be lazy? If so, would calling `getPropertyValues` initialize the lazy value? If this is not an issue, the following looks OK to me.